qemu/tests/qemu_guest_agent.py

```diff
@@ -461,24 +461,34 @@ def gagent_check_get_vcpus(self, test, params, env):
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
         """
-        error_context.context("Check can-offline field of guest agent.", logging.info)
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        error_context.context("Check can-offline field of guest agent.",
+                              logging.info)
         vcpus_info = self.gagent.get_vcpus()
         cpu_num_qga = len(vcpus_info)
         for vcpu in vcpus_info:
             if params.get("os_type") == "linux":
-                if vcpu["logical-id"] == 0 and vcpu["can-offline"] is True:
-                    test.fail("The first logical vcpu can't be offline.")
+                if vcpu["logical-id"] == 0:
+                    vcpu_can_offline_qga = vcpu["can-offline"]
```

Comment:
@xiagao How to get vcpu["can-offline"]'s value? In other words,  it's value come from where?

Reply:
The value can be get from qga command..
And can get the value from guest, like the following command.
cmd = "find /sys/devices/system/cpu/cpu0/ -name online"

Reply:
@xiagao I still don't find this value yet.   e.g  vcpu["can-offline"]=xxxxxx

Reply:
From qga result.
 {"execute":"guest-get-vcpus"}
{"return": [{"online": true, "can-offline": false, "logical-id": 0}, {"online": true, "can-offline": false, "logical-id": 1}, {"online": true, "can-offline": false, "logical-id": 2}, {"online": true, "can-offline": false, "logical-id": 3}]}

Reply:
Acked by: Xiangchun fu xfu@redhat.com